source/test-suite/Main.hs

```diff
@@ -1544,8 +1544,18 @@ main = Hspec.hspec . Hspec.parallel . Hspec.describe "cabal-gild" $ do
       Directory.createDirectory "N"
       writeFile (FilePath.combine "N" "M1.hs") ""
       writeFile (FilePath.combine "N" "M2.hs") ""
-      Gild.mainWith ["--input=i.cabal", "--output=o.cabal"]
-      readFile "o.cabal"
+      Gild.mainWith ["--input=i.cabal", "--output=r.cabal"]
+      readFile "r.cabal"
+        `Hspec.shouldReturn` unlines
+          [ "library",
+            "  -- cabal-gild: discover --exclude=.\\M2.hs --exclude=N/M1.hs",
+            "  exposed-modules:",
+            "    M1",
+            "    N.M2"
+          ]
+      d <- Directory.getCurrentDirectory
+      Gild.mainWith ["--input", FilePath.combine d "i.cabal", "--output=r.cabal"]
```

Comment:
I originally had a pure test for this, but I think it makes more sense to test using the real file system. Windows paths are weird, and using the actual filesystem exercises the functionality without jumping through a bunch of hoops.